api/graphql/models/album.go

```diff
@@ -83,26 +83,28 @@ func GetParentsFromAlbums(db *gorm.DB, filter func(*gorm.DB) *gorm.DB, albumID i
 func (a *Album) Thumbnail(db *gorm.DB) (*Media, error) {
 	var media Media
 
-	if a.CoverID == nil {
-		if err := db.Raw(`
-			WITH recursive sub_albums AS (
-				SELECT * FROM albums AS root WHERE id = ?
-				UNION ALL
-				SELECT child.* FROM albums AS child JOIN sub_albums ON child.parent_album_id = sub_albums.id
-			)
-
-			SELECT * FROM media WHERE media.album_id IN (
-				SELECT id FROM sub_albums
-			) AND media.id IN (
-				SELECT media_id FROM media_urls WHERE media_urls.media_id = media.id
-			) ORDER BY id LIMIT 1
-		`, a.ID).Find(&media).Error; err != nil {
-			return nil, err
-		}
-	} else {
-		if err := db.Where("id = ?", a.CoverID).Find(&media).Error; err != nil {
+	if a.CoverID != nil {
+		if err := db.First(&media, *a.CoverID).Error; err != nil {
 			return nil, err
 		}
+		return &media, nil
+	}
+
+	query := `
+		WITH RECURSIVE sub_albums AS (
+			SELECT id FROM albums WHERE id = ?
+			UNION ALL
+			SELECT a.id FROM albums a
```

Comment:
Please also update table aliases in the recursive querying. 

I suggest using `albums AS children` instead of alias `a`, and keeping `sub_albums` with the original name. `child.parent_album_id = sub_albums.id` is more readable.